src/timing-utilities.js

```diff
@@ -28,18 +28,89 @@
     return clone;
   }
 
-  function makeTiming(timingInput, forGroup) {
-    var timing = {
-      delay: 0,
-      endDelay: 0,
-      fill: forGroup ? 'both' : 'none',
-      iterationStart: 0,
-      iterations: 1,
-      duration: forGroup ? 'auto' : 0,
-      playbackRate: 1,
-      direction: 'normal',
-      easing: 'linear',
-    };
+  function AnimationEffectTiming() {
+    this._delay = 0;
+    this._endDelay = 0;
+    this._fill = 'none';
+    this._iterationStart = 0;
+    this._iterations = 1;
+    this._duration = 0;
+    this._playbackRate = 1;
+    this._direction = 'normal';
+    this._easing = 'linear';
+  }
+
+  AnimationEffectTiming.prototype = {
+    _setMember: function(member, value) {
+      this['_' + member] = value;
+      if (this._effect) {
+        this._effect._timingInput[member] = value;
+        this._effect._timing = shared.normalizeTimingInput(shared.normalizeTimingInput(this._effect._timingInput));
+        this._effect.activeDuration = shared.calculateActiveDuration(this._effect._timing);
+        if (this._effect._animation) {
+          this._effect._animation._rebuildUnderlyingAnimation();
+        }
+      }
+    },
+    set delay(value) {
+      this._setMember('delay', value);
+    },
+    get delay() {
+      return this._delay;
+    },
```

Comment:
You could use Object.defineProperty(AnimationEffectTiming.prototype, name, {
  get: function() { return this['_' + name]; },
  set: function(value) { this._setMember(name, value); }
}
and iterate over the timing properties ('delay', 'endDelay', 'fill', etc). Then it'll be easier to see that all of the properties are handled identically.